packages/testsweets/lib/src/ui/widget_capture/widget_capture_widgets/connection_painter.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:testsweets/src/constants/app_constants.dart';
+
+import 'package:testsweets/src/enums/widget_type.dart';
+import 'package:testsweets/src/extensions/widget_type_extension.dart';
+
+class ConnectionPainter extends CustomPainter {
+  final WidgetType sourcePointType;
+  final Offset sourcePoint;
+  final List<Offset> targetPoints;
+  const ConnectionPainter({
+    required this.sourcePointType,
+    required this.sourcePoint,
+    required this.targetPoints,
+  });
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    canvas.saveLayer(Rect.largest, Paint());
+
+    for (var targetPoint in targetPoints) {
+      final linePaint = Paint()
+        ..color = sourcePointType.getColorOfWidgetType
+        ..strokeWidth = 2;
+
+      canvas.drawLine(sourcePoint, targetPoint, linePaint);
+      final circlePaint = Paint()
+        ..color = Colors.transparent
+        ..blendMode = BlendMode.clear;
+      canvas.drawCircle(
+          targetPoint, WIDGET_DESCRIPTION_VISUAL_SIZE / 2, circlePaint);
+    }
+    final circlePaint = Paint()
+      ..color = Colors.transparent
+      ..blendMode = BlendMode.clear;
+    canvas.drawCircle(
+        sourcePoint, WIDGET_DESCRIPTION_VISUAL_SIZE / 2, circlePaint);
+    canvas.restore();
+  }
+
+  @override
+  bool shouldRepaint(covariant CustomPainter oldDelegate) {
+    return false;
```

Comment:
shouldn't this be checking the position of the widgets and then repaint. 

To see if that's the case you can attach and then move the attached widget after the line has been connected.

Reply:
That's right, but I don't wanna waste time on this functionality cause while I was developing this I had a couple of scenarios where this implementation will not work in the long run
So instead I hide the connections while I move the widget